src/ott/neural/flows/genot.py

```diff
@@ -0,0 +1,490 @@
+      rescaling_b: Callable[[jnp.ndarray], float] = None,
+      unbalanced_kwargs: Dict[str, Any] = types.MappingProxyType({}),
+      callback_fn: Optional[Callable[[jnp.ndarray, jnp.ndarray, jnp.ndarray],
+                                     Any]] = None,
+      rng: Optional[jax.Array] = None,
+  ):
+    rng = utils.default_prng_key(rng)
+    rng, rng_unbalanced = jax.random.split(rng)
+    BaseNeuralSolver.__init__(
+        self, iterations=iterations, valid_freq=valid_freq
+    )
+    ResampleMixin.__init__(self)
+    UnbalancednessMixin.__init__(
+        self,
+        rng=rng_unbalanced,
+        source_dim=input_dim,
+        target_dim=input_dim,
+        cond_dim=cond_dim,
+        tau_a=tau_a,
+        tau_b=tau_b,
+        rescaling_a=rescaling_a,
+        rescaling_b=rescaling_b,
+        unbalanced_kwargs=unbalanced_kwargs,
+    )
+    if isinstance(
+        ot_solver, gromov_wasserstein.GromovWasserstein
+    ) and epsilon is not None:
+      raise ValueError(
+          "If `ot_solver` is `GromovWasserstein`, `epsilon` must be `None`. " +
+          "This check is performed to ensure that in the (fused) Gromov case " +
+          "the `epsilon` parameter is passed via the `ot_solver`."
+      )
+
+    self.rng = utils.default_prng_key(rng)
+    self.velocity_field = velocity_field
+    self.state_velocity_field: Optional[TrainState] = None
+    self.flow = flow
+    self.time_sampler = time_sampler
+    self.optimizer = optimizer
+    self.checkpoint_manager = checkpoint_manager
+    self.latent_noise_fn = jax.tree_util.Partial(
+        jax.random.multivariate_normal,
+        mean=jnp.zeros((output_dim,)),
+        cov=jnp.diag(jnp.ones((output_dim,)))
+    )
+    self.input_dim = input_dim
+    self.output_dim = output_dim
+    self.cond_dim = cond_dim
+    self.k_samples_per_x = k_samples_per_x
+
+    # OT data-data matching parameters
+    self.ot_solver = ot_solver
+    self.epsilon = epsilon
+    self.cost_fn = cost_fn
+    self.scale_cost = scale_cost
+    self.fused_penalty = fused_penalty
+
+    # OT latent-data matching parameters
+    self.solver_latent_to_data = solver_latent_to_data
+    self.kwargs_solver_latent_to_data = kwargs_solver_latent_to_data
+
+    # callback parameteres
+    self.callback_fn = callback_fn
+    self.setup()
+
+  def setup(self):
+    """Set up the model.
+
+    Parameters
+    ----------
+    kwargs
+    Keyword arguments for the setup function.
+    """
+    self.state_velocity_field = (
+        self.velocity_field.create_train_state(
+            self.rng, self.optimizer, self.output_dim
+        )
+    )
+    self.step_fn = self._get_step_fn()
+    if self.solver_latent_to_data is not None:
+      self.match_latent_to_data_fn = self._get_sinkhorn_match_fn(
+          ot_solver=self.solver_latent_to_data,
+          **self.kwargs_solver_latent_to_data
+      )
+    else:
+      self.match_latent_to_data_fn = lambda key, x, y, **_: (x, y)
```

Comment:
I don't think there's a need to define this function, but if you wanna keep it, would change it to:
```python
lambda _, x, y, **__: (x, y)
```